framework/src/utils/PetscSupport.C

```diff
@@ -612,10 +610,9 @@ storePetscOptions(FEProblemBase & fe_problem, const InputParameters & params)
   }
 
   // The parameters contained in the Action
-  const MultiMooseEnum & petsc_options = params.get<MultiMooseEnum>("petsc_options");
-  const MultiMooseEnum & petsc_options_inames = params.get<MultiMooseEnum>("petsc_options_iname");
-  const std::vector<std::string> & petsc_options_values =
-      params.get<std::vector<std::string>>("petsc_options_value");
+  const auto & petsc_options = params.get<MultiMooseEnum>("petsc_options");
+  const auto & petsc_pair_options =
+      params.getPairs<MooseEnumItem, std::string>("petsc_options_iname", "petsc_options_value");
```

Comment:
Should we use `params.get<MooseEnumItem, std::string>` instead of `params.getPairs<MooseEnumItem, std::string>`? 

Because we do not write  `params.get<MultiMooseEnum>`  as  `params.getEnum<MultiMooseEnum >`

Any tech barrier?

Reply:
No barrier, this could work.